titiler/endpoints/factory.py

```diff
@@ -1081,3 +1081,580 @@ def point(
             )
 
             return {"coordinates": [lon, lat], "values": values}
+
+
+# ref: https://github.com/python/mypy/issues/5374
+@dataclass  # type: ignore
+class AsyncBaseFactory(metaclass=abc.ABCMeta):
```

Comment:
I'm not sure there is a better solution than copying the whole code from `non-async` classes. Def not great from a maintainer POV!

Reply:
Why not use mixins?

Reply:
because it doesn't solve the property definition order.  The problem here, is there is no `AsyncReader` (like COGReader) available by default so we need to set `reader: Type[AsyncBaseReader]` without default. In rio-tiler we overcome this kind of problem using attrs ... so I guess we could do the same here